packages/synapse-react-client/src/components/DataGrid/StartGridSession.tsx

```diff
@@ -0,0 +1,292 @@
+import { useState, useRef, useEffect } from 'react'
+import { useCreateGridSession } from './useCreateGridSession'
+import { parseQueryInput } from './DataGridUtils'
+import {
+  CreateGridRequest,
+  PostRepoV1GridSessionSessionIdReplicaRequest,
+  GridSession,
+} from '@sage-bionetworks/synapse-client'
+import { SynapseClient, TableQuery } from '@sage-bionetworks/synapse-client'
+import { useSynapseContext } from '@/utils/context/SynapseContext'
+
+export interface StartGridSessionProps {
+  onSessionChange?: (sessionId: string) => void
+  onReplicaChange?: (replicaId: number | null) => void
+  onPresignedUrlChange?: (url: string) => void
+}
+
+// Based on user input, start a new session with or without a SQL query
+// or join an existing session by ID
+export const StartGridSession = ({
+  onSessionChange,
+  onReplicaChange,
+  onPresignedUrlChange,
+}: StartGridSessionProps) => {
+  const [gridSql, setGridSql] = useState('')
+  const [sessionId, setSessionId] = useState('')
+  const [replicaId, setReplicaId] = useState<number | null>(null)
+  const [presignedUrl, setPresignedUrl] = useState('')
+  const [showDropdown, setShowDropdown] = useState(false)
+  const [availableSessions, setAvailableSessions] = useState<GridSession[]>([])
+
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const synapseClient = useSynapseContext().synapseClient
+
+  const deleteSession = async (sessionId: string) => {
+    try {
+      await synapseClient.gridServicesClient.deleteRepoV1GridSessionSessionId({
+        sessionId,
+      })
+      console.log(`Session ${sessionId} deleted successfully.`)
+      setSessionId('')
+      onSessionChange?.('')
+      setReplicaId(null)
+      onReplicaChange?.(null)
+      setPresignedUrl('')
+      onPresignedUrlChange?.('')
+    } catch (error) {
+      console.error(`Failed to delete session ${sessionId}:`, error)
+    }
+  }
+
+  const getSessionList = async (
```

Comment:
Replace `getSessionList` with a new function `useGetGridSessionsInfinite` in new file `useGridSession.ts`

See `useGetOAuthClientInfinite` as an example